exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverStatsExporter.java

```diff
@@ -392,10 +396,13 @@ private static void createInternal(
       throws IOException {
     synchronized (monitor) {
       checkState(instance == null, "Stackdriver stats exporter is already created.");
-      MetricServiceClient client =
-          stub == null
-              ? createMetricServiceClient(credentials, deadline)
-              : MetricServiceClient.create(stub);
+      final MetricServiceClient client;
```

Comment:
How about saving the `MetricServiceClient` in both cases (stub as well as real)? Closing the stub client seems innocuous.

Reply:
I thought about that as well. 

However, if the user configures a stub they also use in another place too, the behavior could be very surprising. I like the pattern, where the piece of code creating the client will be responsible for closing it, too. Closing the given client would violate that pattern.